assets/javascripts/discourse/controllers/preferences-activity-pub.js

```diff
@@ -2,24 +2,52 @@ import { tracked } from "@glimmer/tracking";
 import Controller from "@ember/controller";
 import { action } from "@ember/object";
 import { notEmpty } from "@ember/object/computed";
-import { ajax } from "discourse/lib/ajax";
-import { popupAjaxError } from "discourse/lib/ajax-error";
+import { service } from "@ember/service";
+import I18n from "I18n";
+import ActivityPubAuthorization from "../models/activity-pub-authorization";
 
 export default class PreferencesActivityPubController extends Controller {
+  @service dialog;
   @tracked authorizations = null;
+  @tracked error;
   @notEmpty("authorizations") hasAuthorizations;
 
+  queryParams = ["error"];
+
+  clearError() {
+    this.error = null;
+  }
+
+  showError() {
+    if (this.error) {
+      this.dialog.alert();
```

Comment:
The empty alert shouldn't be necessary here, right?

Reply:
Done: https://github.com/angusmcleod/discourse-activity-pub/commit/fcc6cd9185f29f719b8f7c1d5130d0c2f8b6f087